File: cpp/leetcode/228.cpp
class Solution {
public:
    vector<string> summaryRanges(vector<int>& nums) {
        vector<string> res;
        for(int i = 0; i < nums.size();){
            int start = i, end = i;
            while(end + 1 < nums.size() && nums[end + 1] == nums[end] + 1) end++;
            if(end > start) res.push_back(to_string(nums[start]) + "->" + to_string(nums[end]));
            else res.push_back(to_string(nums[start]));
            i = end + 1;
        }
        return res;
    }
};

File: python/leetcode/228.py
class Solution(object):
    def summaryRanges(self, nums):
        res, i = [], 0 
        if len(nums) == 1: return [str(nums[0])]
        while i < len(nums):
            start = nums[i]
            while i + 1 < len(nums) and nums[i] == nums[i+1] - 1: i += 1
            if start != nums[i]:
                res.append(str(start) + "->" + str(nums[i]))
            else:
                res.append(str(start))
            i += 1
        return res